test_runner/src/main/kotlin/ftl/args/ArgsHelper.kt

```diff
@@ -178,9 +178,17 @@ object ArgsHelper {
 
     fun calculateShards(filteredTests: List<String>, args: IArgs): List<List<String>> {
         val oldTestResult = GcStorage.downloadJunitXml(args) ?: JUnitTestResult(mutableListOf())
-        val shardsByTime = Shard.calculateShardsByTime(filteredTests, oldTestResult, args)
 
-        return testMethodsAlwaysRun(shardsByTime.stringShards(), args)
+        var shards: MutableList<MutableList<String>>
+
+        try {
+            shards = Shard.createShardsByShardTime(filteredTests, oldTestResult, args)
```

Comment:
This can be improved. After applying this algo to my apk's with `shardTime=110`, the run ended up creating: `Shard times: 40s, 106s, 107s, 108s, 108s, 108s, 109s`

Another possibility can be running `Shard.createShardsByShardTime` to get the shard count, and using that value in `Shard.createShardsByShardCount`. That will make tests more distributed across shards.